llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -808,6 +835,20 @@ bool SPIRVInstructionSelector::selectExtInst(Register ResVReg,
   return false;
 }
 
+bool SPIRVInstructionSelector::selectNAryOpWithSrcs(Register ResVReg,
+                                                    const SPIRVType *ResType,
+                                                    MachineInstr &I,
+                                                    std::vector<Register> Srcs,
+                                                    unsigned Opcode) const {
+  auto MIB = BuildMI(*I.getParent(), I, I.getDebugLoc(), TII.get(Opcode))
+                 .addDef(ResVReg)
+                 .addUse(GR.getSPIRVTypeID(ResType));
+  for (Register SReg : Srcs) {
+    MIB.addUse(SReg);
```

Comment:
I'm pretty sure this function could just replace `selectUnOpWithSrc` because unary wone just be one iteration of the loop. Maybe it would be better to do an NFC change that unifies to one function called `selectOpWithSrcs`?

Reply:
Should I do that as part of this PR?

Reply:
You can do it as a follow on.